doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+applied. Note that if any dependency issue occurs, warnings will be generated
+(e.g. not enabling the configuration of a module via Kconfig).
+
+## Application configuration with Kconfig                 {#app-config-kconfig}
+To expose application-specific configuration options a `Kconfig` file can
+be placed in the application's folder. For an example of this you can check
+the `tests/kconfig` application.
+
+## A note on the usage of CFLAGS
+When a certain module is being configured via Kconfig the configuration macro
+will not longer be overridable by means of CFLAGS (e.g. set on the
+compilation command or on a Makefile). Consider this if you are getting a
+'redefined warning'.
+
+## A note on the usage of the 'clean' command
+When using Kconfig as the configurator for RIOT, configuration symbols may be
+used in Makefiles through the build system. For this to work properly make
+sure that when cleaning an application you call `make clean && make all`,
+instead of both targets on a single command.
+
+---
+# Integration into the build system    {#kconfig-integration-into-build-system}
+
+The integration of Kconfig into the build system is mainly done in
+`makefiles/kconfig.mk`.
+
+## Steps during the build process                {#kconfig-steps-build-process}
+
+![Ouput of every step](kconfig_integration.svg)
+
+### 0. Module dependency resolution
+Currently, the resolution of module dependencies is being performed by the build
+system where all the used modules and packages end up listed in the `USEMODULE`
+make variables. In the next phases of integration we plan to resolve dependencies
+using Kconfig.
+
+
+#### Input
+- Makefiles.
+
+#### Output
+- `USEMODULE` and `USEPKG` variables.
+
+### 1. Module listing
+The list of modules needed for the particular build is dumped into the
+`$ (GENERATED_DIR)/Kconfig.dep` file, where each module is translated into a
+Kconfig symbol as documented in [Appendix A](#kconfig-appendix-a).
+
+#### Input
+- `USEMODULE` and `USEPKG` variables
+
+#### Output
+- `$ (GENERATED_DIR)/Kconfig.dep` file
+
+### 2. Merging all configuration sources
+In this step configuration values are taken from multiple sources and merged
+into a single `merged.config` configuration file. This file is temporary and is
+removed on clean. If the user needs to save a particular configuration
+set, a backup has to be saved (this can be done using the menuconfig interface)
+so it can be loaded later in this step.
+
+To accomplish merging of multiple input files, the `mergeconfig` script is used.
+Note that **the order matters**: existing configuration values are merged in the
+order expressed in the input section, where the last value assigned to a
+parameter has the priority. Any configuration file that is not present will be
+ignored. If no configuration files are available all default values will be
+applied.
+
+`merged.config` is the only configuration input for the `autoconf.h` generation in
+step 4.
```

Comment:
Can you better reference that step in case the order changes some-when in future?